lib/Models/Catalog/Esri/ArcGisFeatureServerCatalogItem.ts

```diff
@@ -483,14 +532,43 @@ export default class ArcGisFeatureServerCatalogItem extends MappableMixin(
     return isDefined(stratum) ? stratum.featureServerData : undefined;
   }
 
-  @computed get mapItems() {
-    if (isDefined(this.geoJsonItem)) {
-      return this.geoJsonItem.mapItems.map(mapItem => {
-        mapItem.show = this.show;
-        return mapItem;
+  /**
+   * Constructs the url for a request to a feature server
+   * @param resultOffset Allows for pagination of results. See https://developers.arcgis.com/rest/services-reference/enterprise/query-feature-service-layer-.htm
+   */
+  buildEsriJsonUrl(resultOffset?: number) {
+    const url = cleanUrl(this.url || "0d");
+    const urlComponents = splitLayerIdFromPath(url);
+    const layerId = urlComponents.layerId;
+
+    if (!isDefined(layerId)) {
+      throw networkRequestError({
+        title: i18next.t(
+          "models.arcGisFeatureServerCatalogItem.invalidServiceTitle"
+        ),
+        message: i18next.t(
+          "models.arcGisFeatureServerCatalogItem.invalidServiceMessage"
+        )
       });
     }
-    return [];
+
+    const uri = new URI(url)
+      .segment("query")
+      .addQuery("f", "json")
+      .addQuery("where", "1=1")
+      .addQuery("outFields", "*")
+      .addQuery("layerDefs", `{"${layerId}": "${this.layerDef}"}`)
+      .addQuery("outSR", "4326");
+
+    if (resultOffset !== undefined) {
+      // Pagination specific parameters
+      uri
+        .addQuery("resultRecordCount", this.featuresPerRequest)
+        .addQuery("orderByFields", "OBJECTID")
```

Comment:
Rather than hardcoding this it can be read from the service metadata as the `objectIdField`.